src/pip/_internal/wheel.py

```diff
@@ -1062,16 +1094,18 @@ def build(
         cache_available = bool(self.wheel_cache.cache_dir)
 
         for req in requirements:
-            ephem_cache = should_use_ephemeral_cache(
+            if not should_build(
+                req,
+                need_wheel=not should_unpack,
+                format_control=format_control,
+            ):
+                continue
+            use_ephemeral_cache = should_use_ephemeral_cache(
```

Comment:
Even if the return value of this function is reversed, I would still keep the sense of the `use_ephemeral_cache` variable to avoid having to change the conditionals below.